tasks/main.yml

```
@@ -7,5 +7,10 @@
     dest: "{{ item.to }}"
     force: yes
     repo: "{{ item.src }}"
-  with_items: "{{ static_web_folder_to_copy }}"
+  loop: "{{ static_web_folder_to_copy }}"
 
+- name: Restrict .git access to root
+  file:
+    path: "{{ item.to }}/.git"
+    mode: 0700
```

```suggestion
    mode: 0700
    recurse: true
```

---

I don't think so, I tested that. Access to `/.git/HEAD` or `/.git/config` are already forbidden with that change. Also, I don't think every file in `.git` should be executable.

---

According to the documentation the state must be directory for this to work:
```suggestion
    mode: 0700
    state: directory
    recurse: true
```

---

> I don't think so, I tested that. Access to `/.git/HEAD` or `/.git/config` are already forbidden with that change. Also, I don't think every file in `.git` should be executable.

Ok. I guess that's some special logic from nginx.

---

But the directory already exists and its type will not be changed by the task

---

> > I don't think so, I tested that. Access to `/.git/HEAD` or `/.git/config` are already forbidden with that change. Also, I don't think every file in `.git` should be executable.
> 
> Ok. I guess that's some special logic from nginx.

The trick is that `root` owns the files. With the `0700` restriction, the `www-data` user has no access to the directory or anything in it.

---

> But the directory already exists and its type will not be changed by the task

My comment with the state was not a response to your comment, but a correction of my initial comment, because the recurse fact does nothing without the state directory (acording to the docs).